src/game_api/state.hpp

```diff
@@ -316,100 +317,53 @@ struct StateMemory
     /// Returns animation_frame of the correct ushabti
     uint16_t get_correct_ushabti();
     void set_correct_ushabti(uint16_t animation_frame);
-};
-#pragma pack(pop)
-
-StateMemory* get_state_ptr();
 
-struct State
-{
-    static void set_do_hooks(bool do_hooks);
+    static std::pair<float, float> click_position(float x, float y);
+    static std::pair<float, float> screen_position(float x, float y);
 
-    static void set_write_load_opt(bool allow);
+    static std::pair<float, float> get_camera_position();
+    void set_camera_position(float cx, float cy);
+    void warp(uint8_t w, uint8_t l, uint8_t t);
+    void set_seed(uint32_t seed);
+    LiquidPhysicsEngine* get_correct_liquid_engine(ENT_TYPE liquid_type);
 
-    static void init(class SoundManager* sound_manager = nullptr);
-    static void post_init();
+    Entity* find(uint32_t uid);
+};
 
-    static State& get();
+struct State
+{
+    uint8_t unknown1[0x3d0];
+    uint32_t frame_count;
+    uint8_t unknown2[0x1c];
+    PRNG prng;
+    uint8_t unknown3[0x10];
+    StateMemory state;
 
     // Returns the main-thread version of StateMemory*
-    StateMemory* ptr_main() const;
+    static StateMemory* ptr_main();
     // Returns the local-thread version of StateMemory*
-    StateMemory* ptr() const;
-    StateMemory* ptr_local() const;
+    static StateMemory* ptr();
+    static StateMemory* ptr_local();
+    static State* get();
+    static State* get_main();

```

Comment:
What was wrong with the old way of first getting the state itself?
also i hate this approach of `State::ptr()`, wasn't the whole discussion about the fact that we want `state` to not be just StateMemory but the whole heap? 

now there is this weird dangers casting. Not to mention that all of those functions now should have error check if the returned heap is null (since it can very much happen in the local)

Reply:
The casting is just the same as before
https://github.com/spelunky-fyi/overlunky/blob/main/src/game_api/thread_utils.hpp#L13-L42

Reply:
should I move this State thing to another PR or something, because the number of changes with this may be too high I guess (?), and also I guess this may need more discussion than the online compatibility stuff I was working on

Reply:
Yeah I guess it turned out bigger and more controversial than I thought, and I wasn't really requesting you to do those changes here and now anyway, this is just where the conversation ended up.